### PR TITLE
Use Psiphon clientlib

### PR DESCRIPTION
**Pull request by fortuna**
Supersedes https://github.com/Jigsaw-Code/outline-sdk/pull/251/

### CODE REVIEW

**Comment by fortuna** [comment]
@jyyi1 any thouhgts?

File: x/psiphon/psiphon.go [hunk 1]
```diff
@@ -76,162 +77,111 @@ var _ transport.StreamDialer = (*Dialer)(nil)
...
 	return streamConn{netConn}, nil
 }
 
-func newPsiphonConfig(config *DialerConfig) (*psi.Config, error) {
+func getClientPlatform() string {
+	clientPlatformAllowChars := func(r rune) bool {
+		return !unicode.IsSpace(r) && r != '_'
+	}
+	goos := strings.Join(strings.FieldsFunc(runtime.GOOS, clientPlatformAllowChars), "-")
+	goarch := strings.Join(strings.FieldsFunc(runtime.GOARCH, clientPlatformAllowChars), "-")
```
Comment (jyyi1) [inline x/psiphon/psiphon.go:97]: Do we really need to normalize these two strings? It seems all [`GOOS`](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/internal/goos/zgoos_linux.go;l=7) and [`GOARCH`](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/internal/goarch/zgoarch_amd64.go;l=7) are already normalized.

**Review by jyyi1 — APPROVED** [review]
Thanks, the code looks good!

File: x/psiphon/psiphon.go (hunk 1 above)
Comment (fortuna) [reply x/psiphon/psiphon.go:97]: I copied the code from @adam-p in https://github.com/Jigsaw-Code/outline-sdk/pull/251.

I think there's no guarantee that GOOS and GOARCH will comply with the Psiphon format, so it makes sense to always escape.

**Pull request merged**